src/frontend/src/binder/insert.rs

```diff
@@ -187,54 +190,82 @@ impl Binder {
-                let err_msg = match num_target_cols.cmp(&num_value_cols) {
-                    std::cmp::Ordering::Equal => None,
-                    std::cmp::Ordering::Greater => {
-                        if has_user_specified_columns {
-                            // e.g. insert into t (v1, v2) values (7)
-                            Some("INSERT has more target columns than expressions")
-                        } else {
-                            // e.g. create table t (a int, b real)
-                            //      insert into t values (7)
-                            // this kind of usage is fine, null values will be provided
-                            // implicitly.
-                            None
-                        }
-                    }
-                    std::cmp::Ordering::Less => {
-                        // e.g. create table t (a int, b real)
-                        //      insert into t (v1) values (7, 13)
-                        // or   insert into t values (7, 13, 17)
-                        Some("INSERT has more expressions than target columns")
-                    }
-                };
-                if let Some(msg) = err_msg {
-                    return Err(RwError::from(ErrorCode::BindError(msg.to_string())));
-                }
-
+                let values_len = values
+                    .0
+                    .first()
+                    .expect("values list should not be empty")
+                    .len();
                 let values = self.bind_values(values.clone(), Some(expected_types))?;
                 bound_query = BoundQuery::with_values(values);
                 cast_exprs = vec![];
+                values_len
+            }
+        };
+
+        let has_user_specified_columns = !cols_to_insert_by_user.is_empty();
+        let num_target_cols = if has_user_specified_columns {
+            cols_to_insert_by_user.len()
+        } else {
+            cols_to_insert_in_table.len()
+        };
+
+        let (err_msg, default_column_indices) = match num_target_cols.cmp(&bounded_column_nums) {
+            std::cmp::Ordering::Equal => (None, default_column_indices),
+            std::cmp::Ordering::Greater => {
+                if has_user_specified_columns {
+                    // e.g. insert into t (v1, v2) values (7)
+                    (
+                        Some("INSERT has more target columns than expressions"),
+                        None,
+                    )
+                } else {
+                    // e.g. create table t (a int, b real)
+                    //      insert into t values (7)
+                    // this kind of usage is fine, null values will be provided
+                    // implicitly.
+                    let mut new_default_column_indices =
+                        col_indices_to_insert.split_off(bounded_column_nums);
+                    new_default_column_indices.extend(default_column_indices.unwrap_or_default());
```

Comment:
this branch is only gonna get executed when default_columns_indices is None, right? 
if so, this line of code may be redundant